codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/EndpointGenerator.java

```diff
@@ -58,6 +58,8 @@
         serviceTrait = service.getTrait(ServiceTrait.class)
                 .orElseThrow(() -> new CodegenException("No service trait found on " + service.getId()));
         endpointPrefix = serviceTrait.getEndpointPrefix();
+        // Isn't SigV4 trait required for AWS service and isn't SigV4Trait::getName required.
+        // So why the fallback to getArnNamespace?
```

Comment:
New auth types can always be added in the future

Reply:
Yes, but how can you tell now what the fallback should be for future auth types? When the case arises, I'm sure we can deal with the changes then. It seems, I could be wrong and hence this is a question, the fallback is dead code.

Reply:
This comment is related to https://github.com/aws/aws-sdk-js-v3/pull/1835 which says _and fall back to arnNamespace(for non-AWS clients)_ , which I don't understand. Why would non-AWS clients have an arnNamespace? They wouldn't have the `ServiceTrait` at all.

Reply:
That's a fair question. It's probably better to just fail here rather than potentially silently failing